system/atomic_image.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION='''
+---
+module: atomic_image
+short_description: Manage the container images on the atomic host platform
+description:
+    - Manage the container images on the atomic host platform
+    - Allows to execute the commands on the container images
+version_added: "2.2"
+author: "Saravanan KR @krsacme"
+notes:
+    - Host should be support ```atomic``` command
+requirements:
+  - atomic
+options:
+    name:
+        description:
+          - Name of the container image
+        required: True
+        default: null
+    state:
+        description:
+          - The state of the container image.
+          - The state ```latest``` will ensure container image is upgraded to the latest version and forcefully restart container, if running.
+        required: False
+        choices: ["present", "absent", "latest"]
+        default: latest
+    started:
+        description:
+          - Start or Stop the continer
+        required: False
+        choices: ["yes", "no"]
+        default: yes
+'''
+
+EXAMPLES = '''
+
+# Execute the run command on rsyslog container image (atomic run rhel7/rsyslog)
+- atomic_image: name=rhel7/rsyslog state=latest
+
+'''
+
+RETURN = '''
+msg:
+    description: The command standard output
+    returned: always
+    type: string
+    sample: [u'Using default tag: latest ...']
+'''
+
+def do_upgrade(module, image):
+    args = ['atomic', 'update', '--force', image]
+    rc, out, err = module.run_command(args, check_rc=False)
+    if rc != 0: # something went wrong emit the msg
+        module.fail_json(rc=rc, msg=err)
+    elif 'Image is up to date' in out:
+        return False
+
+    return True
+
+
+def core(module):
+    image = module.params['name']
+    state = module.params['state']
+    started = module.params['started']
+    is_upgraded = False
+
+    if state == 'present' or state == 'latest':
+        if state == 'latest':
+            is_upgraded = do_upgrade(module, image)
+
+        if started:
+            args = ['atomic', 'run', image]
+        else:
+            args = ['atomic', 'install', image]
+    elif state == 'absent':
+        args = ['atomic', 'uninstall', image]
+
+    out = {}
+    err = {}
+    rc = 0
+    rc, out, err = module.run_command(args, check_rc=False)
```

Comment:
Since we're screenscraping stdout later, do we need to make sure that the output is in the C locale?  If so, you can add locale setting to either  the ```environ_update``` parameter of run_command or add to the ```module.run_command_environ_update``` dict.

Example:
``` python
lang_env = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C')
 rc, out, err = module.run_command(args, check_rc=False, environ_update=lang_env)
```